docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+#### Connected Streams
 
-#### Flink Job Cluster
+A pattern in Flink where a single operator has two input streams. Connected streams can also be used 
+to implement streaming joins.
 
-A Flink Job Cluster is a dedicated [Flink Cluster](#flink-cluster) that only
-executes a single [Flink Job](#flink-job). The lifetime of the
-[Flink Cluster](#flink-cluster) is bound to the lifetime of the Flink Job. 
-This deployment mode has been deprecated since Flink 1.15.  
+#### Connectors
 
-#### Flink Cluster
+Connectors allow [Flink applications](#(flink)-applications) to read from and write to various external 
+systems. They support multiple formats in order to encode and decode data to match Flink’s data structures.
 
-A distributed system consisting of (typically) one [JobManager](#flink-jobmanager) and one or more
-[Flink TaskManager](#flink-taskmanager) processes.
+#### Dataflow
+
+See [logical graph](#logical-graph).
+
+#### DataStream
+
+This is a collection of data in a Flink application. You can think of them as immutable collections 
+of data that can contain duplicates. This data can either be finite or unbounded.
+
+#### Directed Acyclic Graph (DAG)
+
+This is a graph that is directed and without cycles connecting the other edges. It can be used to 
+conceptually represent a [dataflow](#dataflow) where you never look back to previous events.
+
+#### Dispatcher
+
+This is a component of the [JobManager](#jobmanager) and provides a REST interface to submit Flink 
+applications for execution and starts a new [JobMaster](#jobmaster) for each submitted job. It also 
+runs the Flink web UI to provide information about job executions.
 
 #### Event
 
-An event is a statement about a change of the state of the domain modelled by the
-application. Events can be input and/or output of a stream or batch processing application.
-Events are special types of [records](#Record).
+An event is a statement about a change of the state of the domain modelled by the application. Events
+can be input and/or output of a stream processing application. Events are special types of
+[records](#Record).
+
+#### Event Time
+
+The time when an [event](#event) occurred, as recorded by the device producing (or storing) the event.
+For reproducible results, you should use event time because the result does not depend on when the 
+calculation is performed.
+
+If you want to use event time, you will also need to supply a Timestamp Extractor and Watermark Generator 
+that Flink will use to track the progress of event time.
+
+#### Exactly-once
+
+A fault-tolerance guarantee and data delivery approach where nothing is lost or duplicated. This does 
+not mean that every event will be processed exactly once. Instead, it means that every event will affect 
+the state being managed by Flink exactly once.
 
 #### ExecutionGraph
 
-see [Physical Graph](#physical-graph)
+See [Physical Graph](#physical-graph).
+
+#### Externalized Checkpoint
```

Comment:
In my understanding all checkpoints are externalized by now in the sense that they are relocatable, etc.. Would "Retained Checkpoint" be the better term?

cc @dawidwys

Reply:
I agree "Retained checkpoint" is a better name.

For the other part. No, checkpoints are not relocatable and they are not self contained, especially incremental checkpoints. You can assume that only for savepoints.